bftengine/src/bcstatetransfer/RangeValidationTree.cpp

```diff
@@ -94,23 +105,57 @@ std::string NodeVal::getDecoded() const noexcept {
   }
   return oss_en.str();
 }
-//////////////////////////////// End of NodeVal operations ///////////////////////////////////////
+
+//////////////////////////////// NodeInfo  ///////////////////////////////////
+uint64_t NodeInfo::id(uint8_t level, uint64_t rvb_index) {
+  IdData data(level, rvb_index);
+  return data.id;
+}
 
 std::string NodeInfo::toString() const noexcept {
   std::ostringstream os;
-  os << " [" << level << "," << rvb_index << "]";
+  os << " [" << level() << "," << rvb_index() << "]";
   return os.str();
 }
 
-uint32_t RVBNode::RVT_K{};
+uint64_t NodeInfo::parentRvbIndexFromChild(uint64_t child_rvb_index, uint8_t child_level) {
+  // formula: int((child_rvb_index-1)/(RVT_K^parent_level))*(RVT_K^parent_level)+1
+  uint8_t parent_level = child_level + 1;
```

Comment:
Maybe consider leaving this expression as a comment and do the calculation when calculating the span parent below.

Reply:
removed. thanks.